File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int target = nums.size() / 2;
        std::unordered_map<int, int> hashbrown;
        for (auto num : nums) {
            hashbrown[num]++;
            if (hashbrown[num] > target) {
                return num;
            }
        }
        
        return -1;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        m, i = -1, 0
        for num in nums:
            if i == 0:
                m = num
                i = 1
            elif m == num:
                i += 1
            else:
                i -= 1
                
        return m